java-checks/src/main/java/org/sonar/java/checks/HardCodedCredentialsCheck.java

```diff
@@ -68,12 +70,34 @@ public void visitNode(Tree tree) {
       if (isStringLiteral(variable.initializer()) && isPasswordVariableName(variable.simpleName())) {
         addIssue(tree);
       }
-    } else {
+    } else if (tree.is(Tree.Kind.ASSIGNMENT)) {
       AssignmentExpressionTree assignmentExpression = (AssignmentExpressionTree) tree;
       if (isStringLiteral(assignmentExpression.expression()) && isPasswordVariable(assignmentExpression.variable())) {
         addIssue(tree);
       }
+    } else {
+      if (isSettingPassword((MethodInvocationTree) tree)) {
+        addIssue(tree);
+      }
+    }
+  }
+
+  private boolean isSettingPassword(MethodInvocationTree tree) {
+    List<ExpressionTree> arguments = tree.arguments();
+    return arguments.size() == 2 && argumentsAreLiterals(arguments) && isPassword((LiteralTree) arguments.get(0));
+  }
+
+  private boolean isPassword(LiteralTree argument) {
+    return argument.is(Tree.Kind.STRING_LITERAL) && PASSWORD_VARIABLE_PATTERN.matcher(LiteralUtils.trimQuotes(argument.value())).matches();
+  }
+
+  private boolean argumentsAreLiterals(List<ExpressionTree> arguments) {
+    for (ExpressionTree argument : arguments) {
+      if (!(argument instanceof LiteralTree)) {
```

Comment:
You should not use InstanceOf but Tree.Kinds and in this very case STRING_LITERAL